mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,195 @@
+[P0] SHAP explainers
+     [P0] Summary plot
+"""
+
+from PIL.Image import Image, open as open_image
+
+
+class ImageEvaluationArtifact(EvaluationArtifact):
+
+    @classmethod
+    def save_content_to_file(cls, content, output_artifact_path):
+        assert isinstance(content, Image)
+        content.save(output_artifact_path)
+
+    @classmethod
+    def load_content_from_file(cls, local_artifact_path):
+        return open_image(local_artifact_path)
+
+
+class DefaultEvaluator(ModelEvaluator):
+
+    def can_evaluate(self, model_type, evaluator_config=None, **kwargs):
+        return model_type in ["classifier", "regressor"]
+
+    def _log_image_artifact(self, artifacts, temp_dir, do_plot, run_id, artifact_name, dataset_name):
+        pyplot.clf()
+        do_plot()
+        artifact_file_name = f"{artifact_name}_on_{dataset_name}.png"
+        artifact_file_local_path = temp_dir.path(artifact_file_name)
+        pyplot.savefig(artifact_file_local_path)
+        mlflow.log_artifact(artifact_file_local_path, artifact_file_name)
+        artifacts[artifact_file_name] = \
+            ImageEvaluationArtifact(
+                location=get_artifact_uri(run_id, artifact_file_name),
+                content=ImageEvaluationArtifact.load_content_from_file(artifact_file_local_path)
+            )
+
+    def _log_model_explainality(
+        self, artifacts, temp_dir, model, X, dataset_name, run_id
+    ):
+        explainer = shap.Explainer(model, X)
+        shap_values = explainer(X)
+
+        def plot_summary():
+            shap.plots.beeswarm(shap_values, show=False)
+
+        self._log_image_artifact(artifacts, temp_dir, plot_summary, run_id, 'shap_summary', dataset_name)
+
+        def plot_feature_importance():
+            shap.plots.bar(shap_values, show=False)
+
+        self._log_image_artifact(
+            artifacts, temp_dir, plot_feature_importance, run_id, 'shap_feature_importance', dataset_name
+        )
+
+    def _classifier_compute_metrics_and_compute_and_log_artifacts(
+        self, temp_dir, model, X, y, dataset_name, run_id, evaluator_config
+    ):
+        # TODO: how to get num_classes without extra config ?
+        num_classes = evaluator_config['num_classes']
+        y_pred = model.predict(X)
+
+        is_binomial = (num_classes <= 2)
+
+        metrics = EvaluationMetrics()
+        artifacts = {}
+        metrics['accuracy'] = sk_metrics.accuracy_score(y, y_pred)
+        metrics['example_count'] = len(X)
+
+        # TODO: sum/mean on what data ?
+        #  [P0] Sum: Computes the (weighted) sum of the given values.
+        #  [P0] Mean: Computes the (weighted) mean of the given values.
+
+        if is_binomial:
+            if hasattr(model, 'predict_proba'):
+                y_probs = model.predict_proba(X)
+                y_prob = y_probs[:, 1]
+            else:
+                y_probs = None
+                y_prob = None
+
+            confusion_matrix = sk_metrics.confusion_matrix(y, y_pred)
```

Comment:
Should we normalize the confusion matrix?
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html

@BenWilson2 thoughts on this?

Reply:
I typically normalize it if I have a relatively even class balance. For very unbalanced classification problems (i.e. fraud prediction, churn prediction, outlier classification) I tend not to since some values can be so small that without visualization supporting scientific notation a very small percentage will be rendered illegible. 
That being said, one needs A LOT OF DATA to get numbers that small; something you'll only ever see on a SparkML classifier implementation (e.g., 1e9 label==0, 100 label==1) but even then there are better ways to handle massive class imbalance such as that (SMOTE, for instance). 
Normalizing the Confusion Matrix will make generating visualizations a lot easier, though.

Reply:
Thanks for the comment. That makes perfect sense. Now I'm leaning towards not using normalization for unbalanced classification problems. If users need a normalized confusion matrix, they can define a custom evaluator.

Reply:
@WeichenXu123 We can compute the confusion matrix for multi-class problems too, right? Can we move this out of the `is_binomial` block?